For reference, uncommenting the floating point fields gives us:

```
--- FAIL: TestElasticsearchOutputVsExporterSerialization (0.06s)
    logger.go:146: 2026-05-28T16:06:28.128-0400	INFO	elasticsearch.esclientleg	elasticsearch url: http://127.0.0.1:58475
    logger.go:146: 2026-05-28T16:06:28.129-0400	DEBUG	elasticsearch.esclientleg	ES Ping(url=http://127.0.0.1:58475)
    logger.go:146: 2026-05-28T16:06:28.129-0400	WARN	elasticsearch.esclientleg	Failed to create request net/http: nil Context
    logger.go:146: 2026-05-28T16:06:28.129-0400	DEBUG	elasticsearch.esclientleg	Ping request failed with: net/http: nil Context
    logger.go:146: 2026-05-28T16:06:28.130-0400	DEBUG	elasticsearch.esclientleg	Completed dialing successfully{"network.transport": "tcp", "server.address": "127.0.0.1:58475"}
    logger.go:146: 2026-05-28T16:06:28.131-0400	DEBUG	elasticsearch	doBulkRequest: 1 events have been sent to elasticsearch in 2.338125ms.
    otelconsumer_test.go:673:
        	Error Trace:	/Users/cmackenzie/go/src/github.com/elastic/beats/libbeat/otel/otelconsumer/otelconsumer_test.go:673
        	Error:      	Not equal:
        	            	expected: "2"
        	            	actual  : "2.0"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-2
        	            	+2.0
        	Test:       	TestElasticsearchOutputVsExporterSerialization
        	Messages:   	field "float32_int": Beats=2 OTel=2.0
    otelconsumer_test.go:673:
        	Error Trace:	/Users/cmackenzie/go/src/github.com/elastic/beats/libbeat/otel/otelconsumer/otelconsumer_test.go:673
        	Error:      	Not equal:
        	            	expected: "0"
        	            	actual  : "0.0"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-0
        	            	+0.0
        	Test:       	TestElasticsearchOutputVsExporterSerialization
        	Messages:   	field "zero_float": Beats=0 OTel=0.0
    otelconsumer_test.go:673:
        	Error Trace:	/Users/cmackenzie/go/src/github.com/elastic/beats/libbeat/otel/otelconsumer/otelconsumer_test.go:673
        	Error:      	Not equal:
        	            	expected: "[1.5,2,0]"
        	            	actual  : "[1.5,2.0,0.0]"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-[1.5,2,0]
        	            	+[1.5,2.0,0.0]
        	Test:       	TestElasticsearchOutputVsExporterSerialization
        	Messages:   	field "float_slice": Beats=[1.5,2,0] OTel=[1.5,2.0,0.0]
    otelconsumer_test.go:673:
        	Error Trace:	/Users/cmackenzie/go/src/github.com/elastic/beats/libbeat/otel/otelconsumer/otelconsumer_test.go:673
        	Error:      	Not equal:
        	            	expected: "1"
        	            	actual  : "1.0"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-1
        	            	+1.0
        	Test:       	TestElasticsearchOutputVsExporterSerialization
        	Messages:   	field "float64_int": Beats=1 OTel=1.0
FAIL
FAIL	github.com/elastic/beats/v7/libbeat/otel/otelconsumer	0.363s
FAIL
```

--- a/libbeat/otel/otelconsumer/otelconsumer_test.go
+++ b/libbeat/otel/otelconsumer/otelconsumer_test.go
@@ -605,7 +605,8 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 	otelSrv := httptest.NewServer(otelMockES)
 
 	f := elasticsearchexporter.NewFactory()
-	cfg := f.CreateDefaultConfig().(*elasticsearchexporter.Config)
+	cfg, ok := f.CreateDefaultConfig().(*elasticsearchexporter.Config)
+	require.Truef(t, ok, "elasticsearchexporter config must be of type *elasticsearchexporter.Config")
 	cfg.Endpoints = []string{otelSrv.URL}
 	// Reduce the batch flush timeout so the test does not wait the default 10s.
 	qb := cfg.QueueBatchConfig.Get()
@@ -644,7 +645,7 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 	// Compare raw JSON tokens directly so that integer-vs-float differences are visible.
 	beats := rawJSONFields(t, beatsDoc)
 	otel := rawJSONFields(t, otelDoc)
-	assert.Equalf(t, len(beats), len(otel), "top-level field count differs: beats=%d otel=%d", len(beats), len(otel))
+	assert.Lenf(t, beats, len(otel), "top-level field count differs: beats=%d otel=%d", len(beats), len(otel))
 	for field := range otel {
 		assert.Containsf(t, beats, field, "unexpected field %q in OTel document", field)
 	}
@@ -667,7 +668,7 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 			otelNested := rawJSONFields(t, otel[field])
 			for nestedField, beatsNestedRaw := range beatsNested {
 				otelNestedRaw, ok := otelNested[nestedField]
-				assert.Equalf(t, len(beatsNested), len(otelNested), "mapstr_nested field count differs")
+				assert.Lenf(t, beatsNested, len(otelNested), "mapstr_nested field count differs")
 				assert.True(t, ok, "mapstr_nested.%s missing from OTel document", nestedField)
 				assert.Equal(t, string(beatsNestedRaw), string(otelNestedRaw), "mapstr_nested.%s should be serialized identically", nestedField)
 			}

--- a/libbeat/otel/otelconsumer/otelconsumer_test.go
+++ b/libbeat/otel/otelconsumer/otelconsumer_test.go
@@ -644,6 +644,10 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 	// Compare raw JSON tokens directly so that integer-vs-float differences are visible.
 	beats := rawJSONFields(t, beatsDoc)
 	otel := rawJSONFields(t, otelDoc)
+	assert.Equalf(t, len(beats), len(otel), "top-level field count differs: beats=%d otel=%d", len(beats), len(otel))
+	for field := range otel {
+		assert.Containsf(t, beats, field, "unexpected field %q in OTel document", field)
+	}
 
 	// Compare all scalar and slice fields as raw JSON tokens.
 	for field, beatsRaw := range beats {
@@ -663,6 +667,7 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 			otelNested := rawJSONFields(t, otel[field])
 			for nestedField, beatsNestedRaw := range beatsNested {
 				otelNestedRaw, ok := otelNested[nestedField]
+				assert.Equalf(t, len(beatsNested), len(otelNested), "mapstr_nested field count differs")
 				assert.True(t, ok, "mapstr_nested.%s missing from OTel document", nestedField)
 				assert.Equal(t, string(beatsNestedRaw), string(otelNestedRaw), "mapstr_nested.%s should be serialized identically", nestedField)
 			}

--- a/libbeat/otel/otelconsumer/otelconsumer_test.go
+++ b/libbeat/otel/otelconsumer/otelconsumer_test.go
@@ -584,7 +584,10 @@ func TestElasticsearchOutputVsExporterSerialization(t *testing.T) {
 	require.Len(t, beatsBatch.Events(), 1)
 	beatsBatch.Events()[0], _ = beatsEnc.EncodeEntry(beatsBatch.Events()[0])
 	require.Len(t, beatsGroup.Clients, 1)
-	require.NoError(t, beatsGroup.Clients[0].Publish(ctx, beatsBatch))
+	beatsClient, ok := beatsGroup.Clients[0].(outputs.NetworkClient)
+	require.True(t, ok, "ES output client must implement outputs.NetworkClient")
+	require.NoError(t, beatsClient.Connect(ctx))
+	require.NoError(t, beatsClient.Publish(ctx, beatsBatch))
 
 	var beatsDoc []byte
 	select {

